### PR TITLE
enhance: add resource group declarative api

### PR DESCRIPTION
**Pull request by chyezh**
issue: https://github.com/milvus-io/milvus/issues/30931

- Add param `config` for `create_resource_group` api.

- Add `update_resource_groups` api to implement declarative resource group api.

- Add new example for declarative resource group api.

### CODE REVIEW

**Review by longjiquan — APPROVED** [review]
LGTM

File: examples/resource_group_declarative_api.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+    # rg1 missing two query node.
+    create_resource_group("rg1", 2)
+    list_all_resource_groups()
+    # DEFAULT_RESOURCE_GROUP: 1
+    # _PENDING_NODES_RESOURCE_GROUP: 0
+    # rg1: 0(missing 2)
+
+    # scale_out(2)
+    # scale out two new query node into cluster by orchestration system, these node will be added to rg1 automatically.
```
Comment (longjiquan) [inline examples/resource_group_declarative_api.py:90]: These two comments are a little confusing.

Comment (chyezh) [reply examples/resource_group_declarative_api.py:90]: `scale_out` operator cannot be done by milvus itself.
Orchestration system like `k8s` is required.

**Comment by chyezh** [comment]
fixed example issue:

[fix: fix example error](https://github.com/milvus-io/pymilvus/pull/2002/commits/2e86e715ee49a88df35461e35fedee359b2c5009)

**Pull request merged**